In [1]:
import numpy as np
import pandas as pd
import statistics
import sys
sys.path.append('./functions/')
import funciones

In [2]:
local_path = "data/Partido1_IACCvsFranca.csv"
visit_path = "data/Partido1_IACCvsFranca_2.csv"
funciones.four_factors(local_path, visit_path)

(0.5526315789473685,
 0.6132075471698113,
 0.15424164524421594,
 0.1271186440677966,
 0.20689655172413793,
 0.896551724137931,
 0.22807017543859648,
 0.3018867924528302)

In [3]:
funciones.show_four_factor_tabla(funciones.four_factors(local_path, visit_path))

           shooting  turnover  rebounding  Free throws
offensive  0.552632  0.154242    0.206897     0.228070
defensive  0.613208  0.127119    0.896552     0.301887


# 1. Shooting (Eficiencia de campo - eFG%):
Ofensivo (0.552632): Este valor significa que el equipo tiene una eficiencia de campo ofensiva del 55.26%. Es un buen valor, ya que implica que el equipo está siendo relativamente eficiente al convertir sus tiros, lo cual es positivo.
Defensivo (0.613208): El equipo está permitiendo a sus oponentes una eficiencia de campo defensiva de 61.32%. Esto es más alto de lo ideal, ya que refleja que el equipo defensivamente no está siendo tan efectivo al evitar que los rivales conviertan sus tiros.
# 2. Turnover (Pérdidas de balón - TOV%):
Ofensivo (0.154242): Un valor de 15.42% de pérdidas de balón en ofensiva es bastante bajo, lo cual indica que el equipo está controlando bien el balón, evitando errores que den oportunidades al oponente.
Defensivo (0.127119): Un 12.71% de pérdidas de balón defensivas es también bajo, lo cual significa que el equipo está haciendo un buen trabajo forzando a los rivales a cometer errores, aprovechando bien las oportunidades para recuperar el balón.
# 3. Rebounding (Rebotes - REB%):
Ofensivo (0.206897): El equipo está obteniendo 20.69% de los rebotes ofensivos disponibles. Este es un porcentaje bajo, lo que significa que no están siendo muy efectivos en capturar rebotes ofensivos, lo que limita sus oportunidades de anotar tras un fallo.
Defensivo (0.896552): Un 89.66% de los rebotes defensivos es un porcentaje excelente, lo que indica que el equipo está controlando el cristal defensivo de manera dominante, impidiendo que su oponente tenga segundas oportunidades para anotar.
# 4. Free Throws (Tiros libres - FT%):
Ofensivo (0.228070): El equipo está obteniendo 22.81% de sus puntos a través de tiros libres, lo que no es muy alto, pero sugiere que están aprovechando algunas oportunidades de tiros libres en la ofensiva.
Defensivo (0.301887): El equipo está permitiendo al oponente un 30.19% de sus puntos a través de tiros libres. Este es un valor relativamente alto, lo que indica que están cometiendo demasiadas faltas y permitiendo que el rival convierta en la línea de tiros libres con frecuencia.
# Resumen:
## Ofensivo:
El equipo es relativamente eficiente en tiros, controla bien las pérdidas de balón, pero no es muy bueno capturando rebotes ofensivos.
Aprovechan moderadamente los tiros libres.
## Defensivo:
El equipo permite una alta eficiencia de campo a sus oponentes, pero es muy fuerte en rebotes defensivos.
Fuerzan algunas pérdidas de balón y permiten demasiadas faltas, lo que se traduce en una mayor cantidad de puntos permitidos a través de tiros libres.
En general, el equipo tiene fortalezas defensivas (rebotes, control de pérdidas) pero necesita mejorar en su defensa sobre el tiro del oponente y en evitar faltas. En ofensiva, es relativamente eficiente, pero podría mejorar en el aprovechamiento de rebotes ofensivos y tiros libres.